### PR TITLE
Add args-parser library version 6.0.1.0.

### PR DESCRIPTION
**Pull request by igormironchik**
Specify library name and version:  **args-parser/6.0.1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

I'm the author of this library.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/args-parser/6.0.0.0/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
+        self.copy("*.hpp", src=os.path.join(self._source_subfolder, "Args"), dst=os.path.join("include", "args-parser"))
+
+    def package_id(self):
+        del self.info.settings.compiler
+        self.info.header_only()
```
Comment (madebr) [inline recipes/args-parser/6.0.0.0/conanfile.py:51]: The settings object is reset by calling `self.info.header_only`.

https://github.com/conan-io/conan/blob/f230bf104842c160be6ea377497151fe1ea64c38/conans/model/info.py#L575-L578


```suggestion
        self.info.header_only()
```

**Review by madebr — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

**Comment by igormironchik** [comment]
Please, don't approve this PR. I want to change it a little: add possibility to use `find_package()`. Will do it soon, just a few hours. Thanks.

**Review by SSE4 — APPROVED** [review]

**Review by SSE4 — CHANGES_REQUESTED** [review]
author asked not to approve

**Comment by igormironchik** [comment]
@SSE4 Thank you. PR is ready to be reviewed.

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by czoido — APPROVED** [review]

File: recipes/args-parser/6.0.1.0/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
+        self.copy("*.hpp", src=os.path.join(self._source_subfolder, "args-parser"), dst=os.path.join("include", "args-parser"))
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {"args-parser": "args-parser::args-parser"}
+        )
```
Comment (SpaceIm) [inline recipes/args-parser/6.0.1.0/conanfile.py:73]: It doesn't make sense here. This library never had a `args-parser` imported target.
6.0.0.0: `Args`
6.0.0.1: `args-parser::args-parser`

File: recipes/args-parser/6.0.1.0/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+
+conan_basic_setup(TARGETS)
+
+find_package(args-parser REQUIRED)
+
+add_executable(${PROJECT_NAME} example.cpp)
+
+target_link_libraries(${PROJECT_NAME} args-parser)
```
Comment (SpaceIm) [inline recipes/args-parser/6.0.1.0/test_package/CMakeLists.txt:13]: `target_link_libraries(${PROJECT_NAME} args-parse::args-parser)`

**Pull request merged**